src/index.ts

```diff
@@ -112,34 +112,30 @@ export function datadog<TEvent, TResult>(
     currentTraceListener = traceListener;
 
     await traceListener.onStartInvocation(event, context);
-
-    const { result, error, didError } = await traceListener.onWrap(
-      async (localEvent: TEvent, localContext: Context) => {
+    let result: TResult | undefined;
+    try {
+      result = await traceListener.onWrap(async (localEvent: TEvent, localContext: Context) => {
         await metricsListener.onStartInvocation(localEvent);
         if (finalConfig.enhancedMetrics) {
           incrementInvocationsMetric(metricsListener, localContext);
         }
         let localResult: TResult | undefined;
-        let localError: any;
-        let localDidError = false;
         try {
           localResult = (await promHandler(localEvent, localContext)) as TResult | undefined;
         } catch (err) {
           if (finalConfig.enhancedMetrics) {
             incrementErrorsMetric(metricsListener, localContext);
           }
-          localError = err;
-          localDidError = true;
+          throw err;
+        } finally {
+          await metricsListener.onCompleteInvocation();
```

Comment:
Can we test that this function is called independently of a rejection from `promHandler`. Ideally a test that failed before and now passes to improve confidence in the change.

Reply:
Good point, I'll add another test case